documentation/src/main/asciidoc/user_guide/equivalence.adoc

```diff
@@ -1,4 +1,4 @@
-==  Storing objects (e.g. arrays) with custom Equivalence functions
+==  Object Equivalence
```

Comment:
As for fixing the actual contents of the paragraph, I don't want to do it in this PR which is mostly about aesthetics.